lua/entities/gmod_wire_expression2/core/custom/sv_selfaware2.lua

```diff
@@ -1,54 +1,125 @@
 -- Author: Vurv, define runtime functions
 
 --[[
-   _____        __ ____   ___                                        ___ 
-  / ___/ ___   / // __/  /   | _      __ ____ _ _____ _____ ___     |__ \
-  \__ \ / _ \ / // /_   / /| || | /| / // __ `// ___// ___// _ \    __/ /
- ___/ //  __// // __/  / ___ || |/ |/ // /_/ // /   / /   /  __/   / __/ 
-/____/ \___//_//_/    /_/  |_||__/|__/ \__,_//_/   /_/    \___/   /____/ 
+   _____        __ ____   ___                                  ___
+  / ___/ ___   / // __/  /   | _      __ ____ _ _____ ___     |__ \
+  \__ \ / _ \ / // /_   / /| || | /| / // __ `// ___// _ \    __/ /
+ ___/ //  __// // __/  / ___ || |/ |/ // /_/ // /   /  __/   / __/
+/____/ \___//_//_/    /_/  |_||__/|__/ \__,_//_/    \___/   /____/
 
  Adds functions similarly to regular-e2's self-aware core.
 ]]
 
--- Reused from coroutine-core
-local function getE2UDF(compiler,funcname)
-    local funcs = compiler.funcs
-    local e2func = funcs[funcname]
-    if not e2func then -- We will look for any udfs that have the name before the parenthesis.
-        local fncnameproper = funcname:match("(%w*)%(") or funcname
-        for Name,Runtime in pairs(funcs) do
-            local proper = Name:match("(%w*)%(")
-            if proper and proper==fncnameproper then return Runtime end
-        end
-    else
-        return e2func
-    end
-end
+local isfunction, debug_getinfo = isfunction, debug.getinfo
+local string_find, string_sub, table_GetKeys = string.find, string.sub, table.GetKeys
+local luaTableToE2, getE2UDF, getE2Func = vex.luaTableToE2, vex.getE2UDF, vex.getE2Func
 
-local function getE2Func(compiler,funcname) 
-    local funcs = wire_expression2_funcs
-	local e2func = funcs[funcname]
-    if not e2func then -- We will look for any udfs that have the name before the parenthesis.
-        local fncnameproper = funcname:match("(%w*)%(") or funcname
-		for Name,Data in pairs(funcs) do
-            local proper = Name:match("(%w*)%(")
-            if proper and proper==fncnameproper then return Data[3] end -- Return function runtime
-        end
-    else
-        return e2func[3]
-    end
-end
+-- TODO: Set E2 costs ( __e2setcost(N) ).
 
--- Ex: print(defined("print(...)")) or print(defined("e:health()"))
```

Comment:
Just a note, you don't specify `e:` like that, it has to be after the `(` parenthesis.